aws-api-appsync/src/main/java/com/amplifyframework/api/aws/AuthModeStrategyType.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws;
+
+/**
+ * Represents that different auth strategies supported by the client that
+ * interfaces with the AppSync backend.
+ */
+public enum AuthModeStrategyType {
+    /**
+     * Uses the default authorization type from the API configuration
+     * unless the incoming request specifies one.
+     */
+    DEFAULT("default"),
```

Comment:
would `SINGLEAUTH` make sense here?

I'm imagining a switch on this enum type and it looks a bit weird because `default` is a keyword:

```
switch (authModeStrategyType) {
    case MULTIAUTH:
        break;
    case DEFAULT:
        break;
    default:
        // wait another default?!?
}
```